components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ResourceEnvironment.kt

```diff
@@ -0,0 +1,67 @@
+package org.jetbrains.compose.resources
+
+import androidx.compose.runtime.*
+import androidx.compose.ui.LocalSystemTheme
+import androidx.compose.ui.SystemTheme
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.intl.Locale
+
+internal data class ResourceEnvironment(
+    val language: LanguageQualifier,
+    val region: RegionQualifier,
+    val theme: ThemeQualifier,
+    val density: DensityQualifier
+)
+
+@OptIn(InternalComposeApi::class)
+@Composable
+internal fun rememberEnvironment(): ResourceEnvironment {
+    val composeLocale = Locale.current
+    val composeTheme = LocalSystemTheme.current
+    val composeDensity = LocalDensity.current
+
+    //cache ResourceEnvironment unless compose environment is changed
+    //TODO provide top level function with a single cache in a root of compose tree
+    return remember(composeLocale, composeTheme, composeDensity) {
+        ResourceEnvironment(
+            LanguageQualifier(composeLocale.language),
+            RegionQualifier(composeLocale.region),
+            ThemeQualifier.selectByValue(composeTheme == SystemTheme.Dark),
+            DensityQualifier.selectByDensity(composeDensity.density)
+        )
+    }
+}
+
+//expensive operation - do not use during recomposition
+//it is required for a non-composable access to string resources
+internal expect fun getResourceEnvironment(): ResourceEnvironment
```

Comment:
```suggestion
/**
Expensive operation - do not use during recomposition.
It is required for a non-composable access to string resources.
*/
internal expect fun getResourceEnvironment(): ResourceEnvironment
```
I suggest to add doc comment for better comment visibility in IDE.